src/bun.js/readline.exports.js

```diff
@@ -3081,6 +3090,157 @@ function _ttyWriteDumb(s, key) {
+   * @returns {Readline} this
+   */
+  cursorTo(x, y = undefined) {
+    validateInteger(x, "x");
+    if (y != null) validateInteger(y, "y");
+
+    var data = y == null ? CSI`${x + 1}G` : CSI`${y + 1};${x + 1}H`;
+    if (this.#autoCommit) process.nextTick(() => this.#stream.write(data));
+    else ArrayPrototypePush.call(this.#todo, data);
+
+    return this;
+  }
+
+  /**
+   * Moves the cursor relative to its current location.
+   * @param {integer} dx
+   * @param {integer} dy
+   * @returns {Readline} this
+   */
+  moveCursor(dx, dy) {
+    if (dx || dy) {
+      validateInteger(dx, "dx");
+      validateInteger(dy, "dy");
+
+      var data = "";
+
+      if (dx < 0) {
+        data += CSI`${-dx}D`;
+      } else if (dx > 0) {
+        data += CSI`${dx}C`;
+      }
+
+      if (dy < 0) {
+        data += CSI`${-dy}A`;
+      } else if (dy > 0) {
+        data += CSI`${dy}B`;
+      }
+      if (this.#autoCommit) process.nextTick(() => this.#stream.write(data));
+      else ArrayPrototypePush.call(this.#todo, data);
+    }
+    return this;
+  }
+
+  /**
+   * Clears the current line the cursor is on.
+   * @param {-1|0|1} dir Direction to clear:
+   *   -1 for left of the cursor
+   *   +1 for right of the cursor
+   *    0 for the entire line
+   * @returns {Readline} this
+   */
+  clearLine(dir) {
+    validateInteger(dir, "dir", -1, 1);
+
+    var data =
+      dir < 0 ? kClearToLineBeginning : dir > 0 ? kClearToLineEnd : kClearLine;
+    if (this.#autoCommit) process.nextTick(() => this.#stream.write(data));
+    else ArrayPrototypePush.call(this.#todo, data);
+    return this;
+  }
+
+  /**
+   * Clears the screen from the current position of the cursor down.
+   * @returns {Readline} this
+   */
+  clearScreenDown() {
+    if (this.#autoCommit) {
+      process.nextTick(() => this.#stream.write(kClearScreenDown));
+    } else {
+      ArrayPrototypePush.call(this.#todo, kClearScreenDown);
+    }
+    return this;
+  }
+
+  /**
+   * Sends all the pending actions to the associated `stream` and clears the
+   * internal list of pending actions.
+   * @returns {Promise<void>} Resolves when all pending actions have been
+   * flushed to the associated `stream`.
+   */
+  commit() {
+    return new Promise((resolve) => {
+      this.#stream.write(ArrayPrototypeJoin.call(this.#todo, ""), resolve);
+      this.#todo = [];
+    });
+  }
+
+  /**
+   * Clears the internal list of pending actions without sending it to the
+   * associated `stream`.
+   * @returns {Readline} this
+   */
+  rollback() {
+    this.#todo = [];
+    return this;
+  }
+}
+
+var PromisesInterface = class Interface extends _Interface {
+  // eslint-disable-next-line no-useless-constructor
+  constructor(input, output, completer, terminal) {
+    super(input, output, completer, terminal);
+  }
+  question(query, options = kEmptyObject) {
+    return new Promise((resolve, reject) => {
+      var cb = resolve;
+
+      if (options?.signal) {
```

Comment:
Can we do these validations above the function scope and then return Promise.reject?